### PR TITLE
Discover dependencies dynamically

### LINKED ISSUE #396: Windows installer: discover dependencies dynamically
The current Windows installer uses a hard-coded list of dependencies which will need maintenance to deal with new versions. It is also not dynamic when configuration options are changed (eg sound is not enabled).

The list should be based on what the installed programs actually need, as determined with `file(GET_RUNTIME_DEPENDENCIES)` or something similar.

### PR DESCRIPTION
**Pull request by jwrober**
Fixes #396, #337

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -147,12 +103,33 @@ install(
...
+      POST_EXCLUDE_REGEXES "C:[\\\\/][Ww][Ii][Nn][Dd][Oo][Ww][Ss][\\\\/].*"
+      EXECUTABLES
+        "${CMAKE_INSTALL_PREFIX}/freeciv21-*.exe"
+      )
+      message(STATUS "Installing library dependencies for freeciv21 executables...")
+      foreach(_file ${r_deps})
+        file(INSTALL
+          DESTINATION "${CMAKE_INSTALL_PREFIX}"
+          TYPE SHARED_LIBRARY
+          FILES "${_file}"
```
Comment (lmoureaux) [inline CMakeLists.txt:129]: Can't we avoid the loop and pass `${r_deps}` directly here? `FILES` makes me think that this should be possible.

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,51 +34,7 @@ if(WIN32 OR MSYS OR MINGW)
   set(CMAKE_INSTALL_BINDIR ".")
   set(PROJECT_NAME "data")
   set(CMAKE_INSTALL_DOCDIR "${CMAKE_INSTALL_DATAROOTDIR}/doc/")
-
   get_filename_component(MINGW_PATH ${CMAKE_CXX_COMPILER} PATH)
```
Comment (lmoureaux) [inline CMakeLists.txt:37]: I think this line is no longer needed.

**Review by lmoureaux — CHANGES_REQUESTED** [review]
Looks very neat!! With this we won't need to keep track of library versions anymore :v: 

I have two small comments on the code.

File: CMakeLists.txt (hunk 1 above)
Comment (jwrober) [reply CMakeLists.txt:129]: I got error messages when doing this so went with the loop to correct

File: CMakeLists.txt (hunk 2 above)
Comment (jwrober) [reply CMakeLists.txt:37]: Still need the 3 set() and the get_filename_component()

Comment (lmoureaux) [reply CMakeLists.txt:37]: Oh you mean for the Qt plugins? Right...

File: CMakeLists.txt (hunk 1 above)
Comment (lmoureaux) [reply CMakeLists.txt:129]: Fair enough

Comment (jwrober) [reply CMakeLists.txt:129]: second commit forthcoming...

File: CMakeLists.txt [hunk 3]
```diff
@@ -147,12 +103,33 @@ install(
...
+  install(CODE [[
+    message(STATUS "Collecting dependencies for freeciv21 executables...")
+
+    # Take a variable that is available at "install" time and repurpose
+    string(REGEX REPLACE "objdump.exe" "" MINGW_PATH ${CMAKE_OBJDUMP})
+
+    # Function to analyze the third party dll files linked to the exe's
+    #   Uses the repurposed variable from above to tell the function where
+    #   the dll files are located. Ignores dll's that come with Windows.
+    file(GET_RUNTIME_DEPENDENCIES
```
Comment (lmoureaux) [inline CMakeLists.txt:115]: One last comment: this command is new in 3.16 so we need to require it on line 1 (but please keep 3.12 support for Linux, as it's what the Longturn server has).

Comment (jwrober) [reply CMakeLists.txt:127]: Handled in second commit

File: CMakeLists.txt (hunk 2 above)
Comment (jwrober) [reply CMakeLists.txt:46]: yes

**Review by lmoureaux — APPROVED** [review]

**Comment by lmoureaux** [comment]
Ubuntu builds failed:
```
$ cmake . -B build -G Ninja -DCMAKE_INSTALL_PREFIX=$PWD/build/install
CMake Error at CMakeLists.txt:37 (project):
  VERSION not allowed unless CMP0048 is set to NEW
-- Configuring incomplete, errors occurred!
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -152,7 +152,8 @@ if(WIN32 OR MSYS OR MINGW)
     DIRECTORY ${MINGW_PATH}/../share/qt5/plugins/platforms
     DESTINATION ${CMAKE_INSTALL_BINDIR}/plugins
     COMPONENT freeciv21
-    FILES_MATCHING PATTERN *.dll)
+    FILES_MATCHING PATTERN *.dll
+    COMPONENT freeciv21)
```
Comment (lmoureaux) [inline CMakeLists.txt:156]: Isn't `COMPONENT` passed twice now?

Comment (jwrober) [reply CMakeLists.txt:156]: oh crud it is

**Comment by jwrober** [comment]
Ok I'm going to let this go. I think its done.

**Pull request merged**